### ISSUE TITLE
Price product For customer

### ISSUE THREAD

**Issue by DavideDyrecta** [nexcess/magento-turpentine#1369]
I use turpentine with hhvm, i've problem whit block price. I use Group Price, tierprice and promotional rules. The block of price is ANONYMOUSE_*, how can i add name to this block for exclude from caching?

Do you have any suggest?

**Comment by miguelbalparda** [comment]
Have you checked the [wiki](https://github.com/nexcess/magento-turpentine/wiki/ESI_Cache_Policy)?

**Comment by DavideDyrecta** [comment]
Hi friend, yes i've check in the wiki and test all post on git about turpentine and price and block without luck

https://github.com/nexcess/magento-turpentine/issues/202
https://github.com/nexcess/magento-turpentine/issues/1176
https://github.com/nexcess/magento-turpentine/issues/359
https://github.com/nexcess/magento-turpentine/issues/226

**Comment by miguelbalparda** [comment]
If I understand this correctly, you will need to name the block in your layout file or programatically.
`<block type="core/text_list" name="left" as="left"/>` for xml 
`$block = $this->getLayout()->createBlock('customer/form_login')->setTemplate('persistent/customer/form/login.phtml')->setName('blockName');` for code
and then do what those issues are suggesting.

**Comment by DavideDyrecta** [comment]
I will try this solution, now i try with ESI_policy to exclude full content of catalog_product_view without luck. With different group price , the price is the same after any login

**Comment by miguelbalparda** [comment]
Have you checked #838?

**Comment by DavideDyrecta** [comment]
I try this, but doesn't work:

$block = $this->getLayout()->createBlock('catalog/product_price')->setTemplate('catalog/product/price.phtml')->setName('product_price1'); **for code**

No friend, i'm try the https://github.com/nexcess/magento-turpentine/pull/838

**Comment by DavideDyrecta** [comment]
The correct code is:

$block = $this->getLayout()->createBlock('catalog/product_price')->setTemplate('catalog/product/price.phtml')->setNameInLayout('product_price1'); for code

The reference in turpentine is:

<catalog_product_view>
        <reference name="product_price1">
            <action method="setEsiOptions">
                <params>
 <disableEsiInjection>0</disableEsiInjection>
            <method>ajax</method>
            <scope>page</scope>
                   <access>private</access>
                    <registry_keys>
                        <current_category/>
                        <current_layer/>
                        <current_customer/>
                        <current_group/>
                    </registry_keys>
                    <flush_events>
                        <customer_logout/>
                        <customer_login/>
                    </flush_events>
                </params>
            </action>
        </reference>
</catalog_product_view>

But doesn't work

**Comment by miguelbalparda** [comment]
I'm still not sure what are you trying to achieve and how, can you add more details?

**Comment by DavideDyrecta** [comment]
Yes, i've different customers with different price. In catalog_product_view i've the price correctly, but when login with different customer role, the price is the same. I wish the price not being cached in both product_list and in catalog_product_view.

**Comment by DavideDyrecta** [comment]
I ve tested the following code and it's work but with personalization it isn't working. If i exclude page of product with "private" or "customer_group" it doesn't work. 


```
<catalog_product_view>
        <turpentine_cache_flag value="0"/>
</catalog_product_view>
```

**Comment by Stasevi4** [comment]
You can test my solution a had the same problem with price on each customer group 
Changing config file version-3.vcl
```
 if (req.http.X-Varnish-Esi-Access == "customer_group" &&
            req.http.Cookie ~ "customer_group=") {
        hash_data(regsub(req.http.Cookie, "^.*?customer_group=([^;]*);*.*$", "\1"));
    }
```
to
```
 if (req.http.Cookie ~ "customer_group=") {
        hash_data(regsub(req.http.Cookie, "^.*?customer_group=([^;]*);*.*$", "\1"));
    }
```
and apply patch https://github.com/nexcess/magento-turpentine/pull/1252

**Comment by DavideDyrecta** [comment]
Hi i've varnish 4 can it works?

**Comment by Stasevi4** [comment]
I use varnish 3.0.7 and i don't tested with varnish 4 check full file of Esi.php https://github.com/Stasevi4/magento-turpentine/blob/00c36f21187cb95c9bbb6ddd93715a3d7b69a51f/app/code/community/Nexcessnet/Turpentine/Model/Observer/Esi.php#L34-L38
